exporter/exporterhelper/batch_sender.go

```diff
@@ -149,27 +155,39 @@ func (bs *batchSender) send(ctx context.Context, req Request) error {
 // sendMergeSplitBatch sends the request to the batch which may be split into multiple requests.
 func (bs *batchSender) sendMergeSplitBatch(ctx context.Context, req Request) error {
 	bs.mu.Lock()
-	bs.activeRequests.Add(1)
-	defer bs.activeRequests.Add(^uint64(0))
 
 	reqs, err := bs.mergeSplitFunc(ctx, bs.cfg.MaxSizeConfig, bs.activeBatch.request, req)
 	if err != nil || len(reqs) == 0 {
 		bs.mu.Unlock()
 		return err
 	}
+
+	bs.activeRequests.Add(1)
 	if len(reqs) == 1 || bs.activeBatch.request != nil {
 		bs.updateActiveBatch(ctx, reqs[0])
-		batch := bs.activeBatch
-		if bs.isActiveBatchReady() || len(reqs) > 1 {
-			bs.exportActiveBatch()
+		b := bs.activeBatch
+		b.requests++
+		leftover := len(reqs) > 1
+		if bs.isActiveBatchReady() || leftover {
+			bs.exportActiveBatch(func(b *batch) {
```

Comment:
Then the callback will always be the same `bs.activeRequests.Add(-b.requests)` for both cases and can be moved to the exportActiveBatch goroutine

Reply:
Refactored to remove the ugly callback in exportActiveBatch